Add react-datetime w/ npm auto-update

sufuf3 opened the pull request:
Pull request for issue: #10731
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/YouCanBookMe/react-datetime
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/react-datetime
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 21
   - Count of stars: 643
   - Count of forks: 255
 * NPM download stats (optional):
   - Downloads in the last day: 1351
   - Downloads in the last week: 36269
   - Downloads in the last month: 142854

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

sufuf3 commented:
@cdnjs/intern3 Please help me review this pull request, thank you.
Note:
1. There are some files is missing from npm in v0.1.0 and v0.4.1, and I can't get them from GitHub, I'll discuss with the author.
If I get the missing files, I'll add them by sending another pull request.

2. It's a high priority because it has 142,854 downloads in the last month.

extend1994 commented on ajax/libs/react-datetime/package.json:
```diff
@@ -0,0 +1,31 @@
+      "files": [
+        "dist/**/*",
+        "css/**/*",
+        "src/**/*"
```
Please separate these three rules. `dist`, `css` and `src` all should be `basePath`.

sufuf3 replied:
If `dist`, `css`, `src` separate three rules, these files will all in version folder.
But I think it's not a good method because I saw some library in cdnjs is distinguished between folders.
like https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/jquery-ui-bootstrap/0.5pre

extend1994 replied:
OK I understand what you mean. But I think `dist` and `src` (if you have confirmed we need them) should still be basePath.
Ref: https://github.com/extend1994/cdnjs/blob/a2877b2df1c972671018fc42bfa059553333f25d/ajax/libs/slippry/package.json

sufuf3 commented:
@extend1994 I remove the `src`. Although there are some var name and the filename under `src` are the same. I think maybe I can exclude the files which under `src`. ref: https://github.com/YouCanBookMe/react-datetime/blob/master/dist/react-datetime.js#L563-L575

And I modified the `basePath` and let the files all in one folder.

Please help me review this PR, again. Thanks.

extend1994 commented on ajax/libs/react-datetime/package.json:
```diff
@@ -0,0 +1,35 @@
+      ]
+    },
+    {
+      "basePath": "css",
```
`css` is not suitable to be `basePath`. Please Use 
> "basePath": ""
"files": [
 "css/react-datetime*"
]
(Please ignore the indentation here)

sufuf3 commented:
@extend1994 I modified the `package.json` and the commit message I changed it as the followings:
```
Add v0.3.1~0.4.0 of react-datetime manually, cc #10731

Manually add v0.3.1 & v0.4.0 from GitHub due to npm auto-update
can't get the files
```
Please help me review this PR again, thanks.

sufuf3 commented:
@hare1039 I added v2.6.1, please help me review this PR, thanks.

sufuf3 commented:
This is a test please ignore that why I push it again. 
I tested this --> sufuf3/Unitial#1
https://github.com/sufuf3/Unitial/blob/master/gitconfig#L21-L23

sufuf3 commented:
@extend1994 I modified it.
Please help me review this PR again, thanks.

sufuf3 commented:
@extend1994 @clairetsai818 I modified the commit message, please help me review this PR, thanks.

kennynaoh commented:
@sufuf3 So what about `v0.1.0` and `v0.4.1`?

sufuf3 commented:
@kennynaoh I discussed with the author.
It's a high priority because it has 142,854 downloads in the last month.
So, I'll add them by sending another PR when I get the missing files.

kennynaoh commented:
OK

PeterDaveHello commented:
Please don't add map files for css in v0.3.1 next time, as you can see the js file doesn't have it in the same version.

sufuf3 commented:
reply to this comment: https://github.com/cdnjs/cdnjs/pull/11082#issuecomment-297226649
here: https://github.com/YouCanBookMe/react-datetime/issues/292#issuecomment-348010405
> It doesn't look like v0.1.0 nor v0.4.1 were official releases, therefore closing this issue. 
Thanks.
